File: algorithms/ar-prfac/c/prime_factor.c
#include <stdio.h>

void primeNumbers(int num) {
    int i, j, isPrime;
    
     for(i=2; i<=num; i++)
    {
        if(num%i==0)
        {
            isPrime = 1;
            for(j=2; j<=i/2; j++)
            {
                if(i%j==0)
                {
                    isPrime = 0;
                    break;
                }
            }
            if(isPrime==1)
            {
                printf("%d, ", i);
            }
        }
    }
}

int main()
{
    int num;

    printf("Enter any number to print Prime factors: ");
    scanf("%d", &num);

    printf("All Prime Factors of %d are: \n", num);
    primeNumbers(num);

    return 0;
}

File: algorithms/ar-prfac/javascript/prime_factor.js
function primeFactors(num) {
    var number = num
    if (number < 1)
        return;
    for (let counter = 2;
        (counter * counter) <= number; counter++) {
        while (number % counter == 0) {
            console.log(counter);
            number = number / counter;
        }
    }
    if (number > 0) {
        console.log(number);
    }
}

primeFactors(40)